### PR TITLE
setup_pbuilder: point gcc to gcc-7 on trusty

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by tchaikov** [comment]
i will give it a spin locally tomorrow.

**Comment by tchaikov** [comment]
@alfredodeza i think we can pre-install gcc-7 on all centos / trusty jenkins slave for reducing the build time. if it makes sense to you also, i will be happy to update `ansible/slave.yml` also.

**Comment by tchaikov** [comment]
also, i found that all three `setup_pbuilder`s are the same. is it possible to symlink the two of them to one? but i searched the symlinks in the source directory, none of them shows up.
```
find . -type f -ls
```

@alfredodeza is there any reason preventing us from doing so? if not, i'd like do so to dedup this.

File: ceph-dev-build/build/setup_pbuilder [hunk 1]
```diff
@@ -83,6 +83,46 @@ echo "BUILD_HOME=`mktemp -d`" >> ~/.pbuilderrc
...
+OTHERMIRROR="deb http://ppa.launchpad.net/ubuntu-toolchain-r/test/ubuntu $DIST main"
+ALLOWUNTRUSTED=yes
+EXTRAPACKAGES="g++-7"
+HOOKDIR=$hookdir
+EOF
+fi
+
+if [ ! -e $hookdir ]; then
+    mkdir -p $hookdir
+    cat > $hookdir/E10update-gcc-alternatives <<EOF
```
Comment (tchaikov) [inline ceph-dev-build/build/setup_pbuilder]: it looks overkill for using `update-alternatives` in a pbuilder chroot tarball. but actually, i was thinking about reusing this in `slave.yml` . so it's less intrusive when being applied to the jenkins slave node.

**Comment by alfredodeza** [comment]
@tchaikov any changes like this have a big impact on all of our builds. I know there is a lot of duplication, but the build script aren't getting worked on anymore, so it is difficult to try an make improvements that have a major impact.

I don't have any opinions on GCC-7, except maybe for how horrible it is sometimes to rely on PPAs hosted on launchpad. ceph-ansible has had many test failures when it can't retrieve nfs-ganesha from launchpad.

@djgalloway and @ktdreyer you might be interested in these changes

**Comment by tchaikov** [comment]
if the basetgz is cached, and can be reused, we won't suffer that much if the PPA hosted by launchpad is not reachable?

**Comment by alfredodeza** [comment]
@tchaikov caching doesn't mean a lot in this context because our builders are ephemeral. They usually aren't around for more than a couple of hours at the most

**Comment by tchaikov** [comment]
@alfredodeza thanks for the reminder. and sorry, i just keep forgetting this.

**Comment by tchaikov** [comment]
@alfredodeza i added more mirrors for ppa.launchpad.net/ubuntu-toolchain-r. HTH.

**Comment by tchaikov** [comment]
tested locally. ceph builds fine using the pbuilder created with the hook even i change the warning gcc/libstdc++ checking in cmake to a fatal error.

**Comment by tchaikov** [comment]
@liewegas @alfredodeza @djgalloway @ktdreyer anything i can do to make further progress? if you are good with removing support for trusty instead as discussed over ceph-{devel,user} ML. i will try to update ceph-build to do so.

**Comment by tchaikov** [comment]
@alfredodeza thanks for your review. i am merging this then. will fix issues if any pops up.

**Pull request merged**